### PR TITLE
Multihop: replaces the hardcoded values for max_belief_price and max_spread_bps

### PR DESCRIPTION
**Pull request by gangov**
the issue for this PR doesn't mention it, but we also have hard coded the value for max_belief_price, that's why I also added it, too

### CODE REVIEW

**Review by ueco-jb — COMMENTED** [review]
Can you add a testcase which fails with spread exceeding the limit?

**Comment by gangov** [comment]
in regard to the issues that we are experiencing with propagating the error messages between the contracts we created a [small demo](https://github.com/gangov/soroban-error-not-propagating) where this can be reproduced in an swift manner

File: contracts/pool/src/tests/swap.rs [hunk 1]
```diff
@@ -315,6 +315,49 @@ fn test_swap_should_fail_when_referral_fee_is_larger_than_allowed() {
...
+    token1.mint(&user1, &1_001_000);
+    token2.mint(&user1, &2_001_000);
+    pool.provide_liquidity(&user1, &Some(5000), &None, &Some(2_000_000), &None, &None);
+
+    // selling just one token with 1% max spread allowed
+    pool.swap(
+        &user1,
+        &None::<Referral>,
+        &token1.address,
+        &50,
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs:355]: Update this comment.

**Review by ueco-jb — APPROVED** [review]
LGTM

**Pull request merged**